### PR TITLE
FOGL-5475: Add pair system tests to verify north service with north a…

### PR DESCRIPTION
**Pull request by YashTatkondawar**
…nd south python plugins

Signed-off-by: YashTatkondawar <yashtatkondawar@gmail.com>

### CODE REVIEW

File: tests/system/python/pair/test_pyton_north_service.py [hunk 1]
```diff
@@ -0,0 +1,665 @@
...
+            skip_verify_north_interface: Flag for assertion of data using PI web API
+            Assertions:
+                on endpoint GET /fledge/ping
+                on endpoint GET /fledge/south
+                on endpoint GET /fledge/north
+                on endpoint GET /fledge/filter
+                on endpoint GET /fledge/service
+                on endpoint GET /fledge/statistics
+                on endpoint GET /fledge/asset
+                on endpoint GET /fledge/track"""
```
Comment (praveen-garg) [inline tests/system/python/pair/test_pyton_north_service.py:479]: I don't see any

File: tests/system/python/pair/test_pyton_north_service.py [hunk 2]
```diff
@@ -0,0 +1,665 @@
...
+
+        # Wait until south and north services are created and some data is loaded
+        time.sleep(wait_time)
+
+        fledge_url_remote = "{}:8081".format(remote_ip)
+
+        filter_cfg_scale = {"enable": "true"}
+        add_filter("scale", None, "SC1", filter_cfg_scale, fledge_url, local_north_service_name,
```
Comment (praveen-garg) [inline tests/system/python/pair/test_pyton_north_service.py:539]: SC1 declared at module level?

File: tests/system/python/pair/test_pyton_north_service.py [hunk 3]
```diff
@@ -0,0 +1,665 @@
...
+            skip_verify_north_interface: Flag for assertion of data using PI web API
+            Assertions:
+                on endpoint GET /fledge/ping
+                on endpoint GET /fledge/south
+                on endpoint GET /fledge/north
+                on endpoint GET /fledge/filter
+                on endpoint GET /fledge/service
+                on endpoint GET /fledge/statistics
+                on endpoint GET /fledge/asset
+                on endpoint GET /fledge/track"""
```
Comment (praveen-garg) [inline tests/system/python/pair/test_pyton_north_service.py:531]: at least do for /filter and /track

File: tests/system/python/pair/test_pyton_north_service.py [hunk 4]
```diff
@@ -0,0 +1,665 @@
...
```
Comment (praveen-garg) [inline tests/system/python/pair/test_pyton_north_service.py:578]: ?

File: tests/system/python/pair/test_pyton_north_service.py [hunk 5]
```diff
@@ -0,0 +1,665 @@
...
+
+        # Wait until south and north services are created and some data is loaded
+        time.sleep(wait_time)
+
+        fledge_url_remote = "{}:8081".format(remote_ip)
+
+        # Add filter in disbled mode
+        filter_cfg_scale = {"enable": "true"}
+        add_filter("scale", None, "SC1", filter_cfg_scale, fledge_url, local_north_service_name,
```
Comment (praveen-garg) [inline tests/system/python/pair/test_pyton_north_service.py:587]: SC1 at module level?

File: tests/system/python/pair/test_pyton_north_service.py [hunk 6]
```diff
@@ -0,0 +1,665 @@
...
+        assert old_ping_result['dataRead'] < new_ping_result['dataRead']
+        assert old_ping_result_remote['dataRead'] < new_ping_result_remote['dataRead']
+
+        if not skip_verify_north_interface:
+            assert old_ping_result['dataSent'] < new_ping_result['dataSent']
+            assert old_ping_result_remote['dataSent'] < new_ping_result_remote['dataSent']
+            _verify_egress(read_data_from_pi_web_api, pi_host, pi_admin, pi_passwd, pi_db, wait_time, retries,
+                           remote_south_asset_name)
+
+    def test_north_python_service_with_filter_reconfig(self, setup_local, setup_remote, read_data_from_pi_web_api,
```
Comment (praveen-garg) [inline tests/system/python/pair/test_pyton_north_service.py:613]: also please add with filter pipeline reorder and removing filters

File: tests/system/python/conftest.py [hunk 7]
```diff
@@ -145,6 +145,62 @@ def clone_make_install():
         return retval
     return _add_fledge_south
 
+# Change name of variables such as service_name, plugin_type, start_service
+@pytest.fixture
+def add_north():
+    def _add_fledge_north(north_plugin, north_branch, fledge_url, north_instance_name="play", config=None,
```
[the remark this answers is not in the document]
Comment (YashTatkondawar) [reply tests/system/python/conftest.py:151]: By default adding the north instance as a task.

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**